pl_bolts/datamodules/vision_datamodule.py

```diff
@@ -0,0 +1,131 @@
+import os
+from abc import abstractmethod
+from typing import List, Optional, Tuple, Union
+
+import torch
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, Dataset, random_split
+
+
+class VisionDataModule(LightningDataModule):
+
+    EXTRA_ARGS = {}
+    #: Dataset class to use
+    DATASET_CLASS = ...
+    name: str = ""
+    #: A tuple describing the shape of the data
+    DIMS: tuple = ...
+
+    def __init__(
+        self,
+        data_dir: Optional[str] = None,
+        val_split: Union[int, float] = 0.2,
+        num_workers: int = 16,
+        normalize: bool = False,
+        seed: int = 42,
+        batch_size: int = 32,
+        *args,
+        **kwargs,
+    ):
+        """
+        Args:
+            data_dir: Where to save/load the data
+            val_split: Percent (float) or number (int) of samples to use for the validation split
+            num_workers: How many workers to use for loading data
+            normalize: If true applies image normalize
+            seed: Seed to fix the validation split
+            batch_size: How many samples per batch to load
+        """
+
+        super().__init__(*args, **kwargs)
+
+        self.data_dir = data_dir if data_dir is not None else os.getcwd()
+        self.val_split = val_split
+        self.num_workers = num_workers
+        self.normalize = normalize
+        self.seed = seed
+        self.batch_size = batch_size
+
+    def prepare_data(self):
+        """
+        Saves files to data_dir
+        """
+        self.DATASET_CLASS(self.data_dir, train=True, download=True)
+        self.DATASET_CLASS(self.data_dir, train=False, download=True)
+
+    def setup(self, stage: Optional[str] = None):
+        """
+        Creates train, val, and test dataset
+        """
+        if stage == "fit" or stage is None:
+            train_transforms = self.default_transforms() if self.train_transforms is None else self.train_transforms
+            val_transforms = self.default_transforms() if self.val_transforms is None else self.val_transforms
+
+            dataset_train = self.DATASET_CLASS(self.data_dir, train=True, transform=train_transforms, **self.EXTRA_ARGS)
+            dataset_val = self.DATASET_CLASS(self.data_dir, train=True, transform=val_transforms, **self.EXTRA_ARGS)
+
+            # Split
+            self.dataset_train = self._split_dataset(dataset_train)
+            self.dataset_val = self._split_dataset(dataset_val, train=False)
+
+        elif stage == "test" or stage is None:
```

Comment:
I think `stage is None` in `elif` is unnecessary in this case:
```python
if stage== "fit" or stage is None:
    ...
elif stage == "test":
    ...
```

or if we want to setup both "fit" and "test" when `stage is None`,  we should write the conditions with `if` without `elif`:
```python
if stage== "fit" or stage is None:
    ...
if stage == "test" or stage is None:
    ...
```
cc: @Borda

Reply:
good catch